runtime/src/main/java/io/micronaut/retry/intercept/AnnotationRetryStateBuilder.java

```diff
@@ -78,17 +78,21 @@ public RetryState build() {
 
     private static RetryPredicate createPredicate(Class<? extends RetryPredicate> predicateClass, AnnotationValue<Retryable> retry) {
         if (predicateClass.equals(DefaultRetryPredicate.class)) {
-            List<Class<? extends Throwable>> includes = resolveIncludes(retry, INCLUDES);
-            List<Class<? extends Throwable>> excludes = resolveIncludes(retry, EXCLUDES);
-            return new DefaultRetryPredicate(includes, excludes);
+            // annotationClassValues are independent from the runtime classpath
+            AnnotationClassValue<?>[] annotationClassValues = retry.annotationClassValues(INCLUDES);
```

Comment:
At this point if the `annotationClassValues` is not empty, but `includes` is, then you could simply return a predicate that always returns false because the thrown annotation could never be in the includes list.

Reply:
I don't think modifying the API of `DefaultRetryPredicate` is necessary. Let me know if I have misunderstood

Reply:
Thanks, that is correct and probably simplifies things. I changed the PR accordingly.